src/main/java/org/folio/dataexp/service/DataExportService.java

```diff
@@ -2,79 +2,88 @@
+    } catch (DataExportRequestValidationException e) {
+      updateJobExecutionForPostDataExport(jobExecutionEntity, jobProfileEntity, JobExecution.StatusEnum.FAIL);
+      log.error(e.getMessage());
+      return;
     }
-    var jobExecution = new JobExecution();
-    jobExecution.setId(UUID.randomUUID());
-    jobExecutionEntityRepository.save(JobExecutionEntity.builder()
-      .id(jobExecution.getId()).jobExecution(jobExecution).build());
-    fileDefinition.setJobExecutionId(jobExecution.getId());
-    fileDefinition.setStatus(FileDefinition.StatusEnum.NEW);
-    var entity = FileDefinitionEntity.builder()
-      .id(fileDefinition.getId()).creationDate(LocalDateTime.now()).fileDefinition(fileDefinition).build();
-    var saved = fileDefinitionEntityRepository.save(entity);
-    return saved.getFileDefinition();
-  }
+    log.info("Post data export for file definition {} and job profile {} with job execution {}",
+      exportRequest.getFileDefinitionId(), exportRequest.getJobProfileId(), jobExecutionEntity.getId());
 
-  public FileDefinition getFileDefinitionById(UUID fileDefinitionId) {
-    return fileDefinitionEntityRepository.getReferenceById(fileDefinitionId).getFileDefinition();
-  }
+    inputFileProcessor.readFile(fileDefinition);
+    slicerProcessor.sliceInstancesIds(fileDefinition);
 
-  public FileDefinition uploadFile(UUID fileDefinitionId, Resource resource) {
-    try {
-      return fileUploadService.uploadFile(fileDefinitionId, resource);
-    } catch (Exception e) {
-      fileUploadService.errorUploading(fileDefinitionId);
-      throw new UploadFileException(e.getMessage());
-    }
+    updateJobExecutionForPostDataExport(jobExecutionEntity, jobProfileEntity, JobExecution.StatusEnum.IN_PROGRESS);
+    singleFileProcessorAsync.exportBySingleFile(jobExecutionEntity.getId(), exportRequest.getRecordType());
   }
 
-  private boolean isNotValidFileNameExtension(String fileName) {
-    return !FilenameUtils.isExtension(fileName.toLowerCase(), CSV_FORMAT_EXTENSION) && !FilenameUtils.isExtension(fileName.toLowerCase(), CQL_FORMAT_EXTENSION);
+  private void updateJobExecutionForPostDataExport(JobExecutionEntity jobExecutionEntity, JobProfileEntity jobProfileEntity, JobExecution.StatusEnum jobExecutionStatus) {
+    var jobExecution = jobExecutionEntity.getJobExecution();
+    jobExecution.setJobProfileId(jobProfileEntity.getJobProfile().getId());
+    jobExecution.setJobProfileName(jobProfileEntity.getJobProfile().getName());
+    jobExecution.setStatus(jobExecutionStatus);
+    var currentDate = new Date();
+    jobExecution.setStartedDate(currentDate);
+    jobExecution.setLastUpdatedDate(currentDate);
+
+    var user = userClient.getUserById(folioExecutionContext.getUserId().toString());
+    var runBy = new JobExecutionRunBy();
+    runBy.firstName(user.getPersonal().getFirstName());
+    runBy.lastName(user.getPersonal().getLastName());
```

Comment:
Probably it would need to set a new field **userId** to runBy as well after [MDEXP-639](https://issues.folio.org/browse/MDEXP-639)

Reply:
@obozhko-folio , userId field was added.